### PR TITLE
[#175716279][#175716280] Added feature and model tests

### PR DESCRIPTION
**Pull request by miljan-kolosek**

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+   describe "Validations" do
+    it { is_expected.to validate_presence_of(:full_name) }
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_presence_of(:bio) }
+  end
+
+  context 'association' do
+    it { should have_many(:stories)}
+  end
+
```
Comment (tamara-kolosek) [inline spec/models/user_spec.rb:38]: Aren't there any method in User model (and other models) to be tested
For example: pdate_admin, compute slug.. and scopes also

**Review by tamara-kolosek — CHANGES_REQUESTED** [review]

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,39 +1,66 @@
...
+      expect(@user.slug).to eql("Full-N")
+    end
+  end
+
+  describe "#admin" do
+    it "should user has admin role" do
+      expect(@user.role).to eq("admin")
+    end
+  end
+
```
Comment (tamara-kolosek) [inline spec/models/user_spec.rb:65]: when testing methods, test different cases

**Review by tamara-kolosek — CHANGES_REQUESTED** [review]

File: spec/models/story_spec.rb [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_presence_of(:content) }
+  end
+
+  describe 'association' do
+    it { should have_many(:taggings)}
+    it { should have_many(:tags).through(:taggings)}
+  end
+
```
Comment (tamara-kolosek) [inline spec/models/story_spec.rb:23]: havent you method tag_list ?

**Review by tamara-kolosek — CHANGES_REQUESTED** [review]

File: spec/models/story_spec.rb (hunk 3 above)
Comment (miljan-kolosek) [reply spec/models/story_spec.rb:23]: Yes, I have.

File: spec/models/story_spec.rb [hunk 4]
```diff
@@ -21,4 +25,14 @@
     it { should have_many(:tags).through(:taggings)}
   end
 
+  describe "tag_list" do
+    it "should story has tags" do
+      @story.tag_list = "123, 345"
+      expect(@story.tags).to match_array([
+        have_attributes(id: 1, name: '123'),
+        have_attributes(id: 2, name: '345'),
+        ])
```
Comment (tamara-kolosek) [inline spec/models/story_spec.rb:34]: seems like it somehow work. Could you just explain when  tag_list= is beeing called?

**Review by tamara-kolosek — APPROVED** [review]

File: spec/models/story_spec.rb (hunk 4 above)
Comment (miljan-kolosek) [reply spec/models/story_spec.rb:34]: Method "tag_list" is called in form when I insert tags into a story, and make tags like array "tag1, tag2". After that, method tag_list=(tags_string) convert all tags into separate object.

Comment (miljan-kolosek) [reply spec/models/story_spec.rb:34]: Method was called on 30 line of code : ` @story.tag_list = "123, 345"`.

Comment (tamara-kolosek) [reply spec/models/story_spec.rb:34]: OK, I see this is something called, assignment method.

**Review by tamara-kolosek — APPROVED** [review]

**Pull request merged**